types/validator_set.go

```diff
@@ -94,7 +99,13 @@ func (vals *ValidatorSet) ValidateBasic() error {
 		return fmt.Errorf("proposer failed validate basic, error: %w", err)
 	}
 
-	return nil
+	for _, val := range vals.Validators {
+		if bytes.Equal(val.Address, vals.Proposer.Address) {
+			return nil
+		}
+	}
```

Comment:
I'm wondering if we also need to ensure this check in blobstream

I don't think so cause it only messes with the initial state? cc @rach-id more info https://github.com/cometbft/cometbft/security/advisories/GHSA-g5xx-c4hv-9ccc

Reply:
it's an interesting thing.

If I understand correctly, this affects the nodes, in case of a malicious RPC endpoint, where it gets the information about the proposer priority wrong, and ends up choosing different validators than the main chain.

I think that it's fine not to include this. In the light client verification, we're just taking the headers along with the signatures. The prover doesn't care about the proposer priority or tries to catch up with the chain.

Reply:
makes sense, thanks!!